src/v/storage/fs_utils.cc

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "fs_utils.h"
+
+#include "config/node_config.h"
+#include "model/namespace.h"
+
+namespace storage {
+
+ss::sstring segment_full_path::string() const {
+    if (unlikely(override_path.has_value())) {
+        return *override_path;
+    }
+
+    return ss::format(
+      "{}/{}_{}/{}-{}-{}.{}",
+      config::node().data_directory().as_sstring(),
+      dir_part.ntp.path(),
+      dir_part.revision_id,
+      file_part.base_offset(),
+      file_part.term(),
+      to_string(file_part.version),
+      extension);
+}
+
+ss::sstring partition_path::make_string() const {
+    return ss::format(
+      "{}/{}_{}", config::node().data_directory(), ntp.path(), revision_id);
+}
+
+std::optional<segment_full_path> segment_full_path::parse(
+  partition_path const& dir_part, const ss::sstring& filename) noexcept {
+    std::optional<segment_path::metadata> file_part_opt;
+    try {
+        file_part_opt = segment_path::parse_segment_filename(filename);
+    } catch (...) {
+        return std::nullopt;
+    }
+    if (!file_part_opt) {
+        return std::nullopt;
+    }
+
+    return segment_full_path(
+      partition_path(dir_part), std::move(*file_part_opt));
+}
+
+segment_full_path segment_full_path::mock(ss::sstring str_path) {
+    auto ntp = model::ntp(
+      model::kafka_namespace,
+      model::topic_partition(
+        model::topic{"testtopic"}, model::partition_id{0}));
+
+    auto pp = partition_path(ntp, model::revision_id{123});
+
+    auto sp = segment_path::metadata{
+      .base_offset = model::offset{123},
+      .term = model::term_id{456},
+      .version = record_version_type::v1};
+
+    return segment_full_path(str_path, pp, sp);
+}
+
+bool segment_full_path::is_internal_topic() const {
+    return dir_part.ntp.ns != model::kafka_namespace;
```

Comment:
Should we include `__consumer_offsets` topic ?

Reply:
I think we don't need to, because that topic has plenty of raft_data in it, so will get indexed properly.

We _could_ treat it as internal (because it won't have custom timestamps in it), but I think it's simpler to say that everything in `kafka` namespace should be treated like user data, rather than keep track of other pseudo-internal topics (audit, etc) here as we add them.